erpnext/hr/doctype/shift_assignment/shift_assignment.py

```diff
@@ -4,7 +4,76 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
 from frappe.model.document import Document
+from frappe.utils import cint, cstr, date_diff, flt, formatdate, getdate
+
+class OverlapError(frappe.ValidationError): pass
 
 class ShiftAssignment(Document):
-	pass
+	def validate(self):
+		self.validate_overlapping_dates();
+
+	def validate_overlapping_dates(self):
+			if not self.name:
+				self.name = "New Shift Assignment"
+
+			for d in frappe.db.sql("""
```

Comment:
no need for a for loop, no possibility for multiple. Add date condition in query itself.